File: Client/src/app/source/models/dtos/menus/visibility-element-request.ts
export class VisibilityElementRequest {
    constructor(
        public menuId: number|null,
        public categoryId: number|null,
        public productId: number|null,
        public isVisible: boolean|null,
        public isAvailable: boolean|null,
        public isSoldOut: boolean|null
    ) { }
}

File: Server/Source/Models/DTOs/Business/MenuStuff/VisibilityElementRequest.cs
namespace Server.Source.Models.DTOs.Business.MenuStuff
{
    public class VisibilityElementRequest
    {
        public int? MenuId { get; set; }
        public int? CategoryId { get; set; }
        public int? ProductId { get; set; }

        public bool? IsVisible { get; set; }
        public bool? IsAvailable { get; set; }
    }
}
